hera_mc/cm_handling.py

```diff
@@ -160,13 +140,11 @@ def find_ports(self, connection_dossier):
 
         input_ports = []
         output_ports = []
-        for k, v in connection_dossier.iteritems():
-            if k in self.non_class_connection_dossier_entries:
-                continue
-            if ':up:' in k and v.downstream_input_port not in input_ports:
-                input_ports.append(v.downstream_input_port)
-            elif ':down:' in k and v.upstream_output_port not in output_ports:
-                output_ports.append(v.upstream_output_port)
+        for k, c in connection_dossier['connections'].iteritems():
+            if c.downstream_input_port not in input_ports:
+                input_ports.append(c.downstream_input_port)
+            elif c.upstream_output_port not in output_ports:
+                output_ports.append(c.upstream_output_port)
```

Comment:
If I'm interpreting this correctly, you're trying to make sorted lists of unique items here. An easier and more efficient way to do this is to use the `set()` type, which acts like a mathematical set and has an `.add()` method that adds item. Then you can write `input_ports = sorted(input_ports)` to convert the set into a sorted list.